fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePlanner.java

```diff
@@ -229,9 +229,11 @@ private LuceneIndexQueryPlan getComparisonsForLuceneFilter(@Nonnull Index index,
         return null;
     }
 
-    public boolean validateIndexField(@Nonnull Index index,
-                                       @Nonnull String field) {
-        return listIndexFieldNames(index.getRootExpression()).contains(field);
+    public boolean validateIndexField(@Nonnull Index index, @Nonnull String field) {
+        final RecordType recordType = getRecordMetaData().recordTypesForIndex(index).iterator().next();
```

Comment:
Looks this is being used in multiple places. So does it make sense to have this as a method in ```RecordMetaData``` to be reused?

Reply:
I changed these to check all record types, so they are no longer identical.